src/components/editor/Editor.tsx

```diff
@@ -1,24 +1,53 @@
 import React from 'react';
-import AttributeEditor from './AttributeEditor';
-import ResourceProfileMapping from './ResourceProfileMapping';
+import AttributeEditor from './attributeEditor/AttributeEditor';
+import ResourceProfileMapping from './resourceProfileMapping/ResourceProfileMapping';
+import ProfileSettings from './profileSettings/ProfileSettings';
 import { useSelector } from 'react-redux';
 import { RootState } from '../../state/store';
+import { ElementDefinition, String } from "../../resources/ts/proto/r4/core/datatypes_pb";
 
 const Editor: React.FC<{}> = () => {
-    const { loading, profile } = useSelector((state: RootState) => state.resource);
+    const { loading, profile, selectedAttributeId } = useSelector((state: RootState) => state.resource);
+    const profileSettings: String.AsObject = 'profileSettings' as unknown as String.AsObject;
+    
+    let attribute: ElementDefinition.AsObject | null = null;
+
+    if (selectedAttributeId && selectedAttributeId !== profileSettings){
+        profile?.snapshot?.element.map((element: ElementDefinition.AsObject) => {
+            if (element.id === selectedAttributeId){
+                attribute = element;
+            }
+        })
```

Comment:
I don't think Array.map() is the best method to use in this case. It should be used when you need to get a modified array from the original one.
Instead, as you only need to find one element matching your condition, you should use Array.find(). That way you'll get the first element matching your request.